timeseries/src/autogluon/timeseries/models/statsmodels/abstract_statsmodels.py

```diff
@@ -0,0 +1,175 @@
+        to predict. At inference time, this will be the number of time steps the model will
+        predict.
+    name: str
+        Name of the model. Also, name of subdirectory inside path where model will be saved.
+    eval_metric: str
+        objective function the model will be scored on, will use mean_wQuantileLoss by default.
+    hyperparameters:
+        various hyperparameters that will be used by model (can be search spaces instead of
+        fixed values). See *Other Parameters* in each inheriting model's documentation for
+        possible values.
+    """
+
+    statsmodels_allowed_init_args: List[str] = []
+    statsmodels_allowed_fit_args: List[str] = []
+    quantile_method_name: str = "pred_int"
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        name = name or re.sub(r"Model$", "", self.__class__.__name__)
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+        self._fitted_models: Dict[str, FittedLocalModel] = {}
+
+    def _get_default_model_init_and_fit_args(self) -> Tuple[dict, dict]:
+        default_model_init_args = {}
+        default_model_fit_args = {}
+        unused_args = []
+        for key, value in self._get_model_params().items():
+            if key in self.statsmodels_allowed_init_args:
+                default_model_init_args[key] = value
+            elif key in self.statsmodels_allowed_fit_args:
+                default_model_fit_args[key] = value
+            else:
+                unused_args.append(key)
+        if len(unused_args) > 0:
+            logger.warning(
+                f" {self.name} ignores the following hyperparameters: {unused_args}. "
+                f"See the docstring of {self.name} for the list of supported hyperparameters."
+            )
+        return default_model_init_args, default_model_fit_args
+
+    def _fit_local_model(
+        self, timeseries: pd.Series, default_model_init_args: dict, default_model_fit_args: dict
+    ) -> FittedLocalModel:
+        """Fit a single local model to the time series."""
+        raise NotImplementedError
+
+    def _fit(self, train_data: TimeSeriesDataFrame, time_limit: int = None, **kwargs):
+        # Select timeseries that don't have a fitted local model
+        train_hash = hash_ts_dataframe_items(train_data)
+        items_to_fit = [item_id for item_id, ts_hash in train_hash.iteritems() if ts_hash not in self._fitted_models]
+        timeseries_to_fit = (train_data.loc[item_id][self.target] for item_id in items_to_fit)
```

Comment:
According to my understanding and based on the experiments that I ran, joblib always uses `n_jobs` x `ram_for_a_single_process` memory. Even for the largest datasets in our benchmark, `ram_for_a_single_process` doesn't go above 300-400MB. This seems to be very manageable since we default to using 50% of the available cores. E.g., even with only 4GB of RAM available to AGTS, we will only run into problems if a machine has 20 cores, which is very unrealistic.

As for your original question, using lists or generators doesn't seem to impact the performance (the generator is slightly faster).

Reply:
The extra memory for each job doesn't change if we train multiple local models, e.g., training `ETS` or `ETS + ARIMA` takes the same amount of RAM in total.